### PR TITLE
Add `oBrandOverride` and update `oBrandGet`.

### PR DESCRIPTION
**Pull request by notlee**
Adds `oBrandOverride` to override component configuration for the current brand.
Updates `oBrandGet` to explicitly retrieve a variant variable within a function (where it is not possible to use `oBrandConfigureFor`).

I'm following up with larger README changes based on @gvonkoss's feedback to make things easier to follow.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,6 +179,14 @@ Nesting is also supported:
...
+
+```scss
+	.o-example--inverse {
+		content: oBrandGet('o-example', 'component-content', $force-variant: 'inverse'); // "inverse" variant value
+	}
+```
+
 ### Output Styles Only If A Brand Supports A Variant
 
 Not all brands will share variants. Define support in the `settings` map as demonstrated above. To output styles only if the current brand supports the variant use `oBrandConfigureFor`.
```
Comment (gvonkoss) [inline README.md]: You said you'd update the README after this, but just noting that 'above' isn't applicable anymore

File: README.md [hunk 2]
```diff
@@ -208,6 +217,28 @@ Uses of `oBrandGet` within `oBrandConfigureFor` [retrieves a variable for a vari
 }
 ```
 
+### Override The Current Brand
+
+It may be desirable to output a customised variant of a component based on brand configuration. To do this create a component specific mixin which maps arguments to `oBrandOverride`. This prevents brand configuration from becoming a public interface, which makes it possible to change brand variables at a later point without a breaking change.
```
Comment (gvonkoss) [inline README.md]: Could we reword this to something along lines of, 'If a variant requires customisation beyond brand configuration, it can be applied through a  component specific mixin.... ' ?

File: README.md [hunk 3]
```diff
@@ -208,6 +217,28 @@ Uses of `oBrandGet` within `oBrandConfigureFor` [retrieves a variable for a vari
...
+
+The following contrived example shows how to override brand variables:
+
+```scss
+@mixin oExample($background-color, $forground-color) {
```
Comment (gvonkoss) [inline README.md]: Perhaps this functionality would be more clear if we named the contrived mixin something bordering on painfully explicit, like `@mixin oExampleSpecifyCustomizedAccentColors`? (I know, sooper excessive, but mostly just trying to illustrate a point)

File: scss/mixins.scss [hunk 4]
```diff
@@ -111,36 +107,112 @@ $_o-brand-depth: () !default; // The number of current variant parts by componen
...
+/**
+* Validate config.
+*
+* @access private
+* @param {string} $component
+* @param {string} $brand
+* @param {map} $config
+* @return {map}
+**/
+@function _oBrandValidateConfig($component, $brand, $config) {
```
Comment (gvonkoss) [inline scss/mixins.scss]: I like this!

File: README.md (hunk 3 above)
Comment (notlee) [reply README.md]: _Maybe_ not quite that ;) but yeah good idea 👍

**Comment by notlee** [comment]
Will follow up with the readme changes :)
